packages/twenty-server/src/workspace/workspace-health/workspace-health.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+
+import { WorkspaceHealthIssue } from 'src/workspace/workspace-health/interfaces/workspace-health-issue.interface';
+import {
+  WorkspaceHealthMode,
+  WorkspaceHealthOptions,
+} from 'src/workspace/workspace-health/interfaces/workspace-health-options.interface';
+
+import { TypeORMService } from 'src/database/typeorm/typeorm.service';
+import { DataSourceService } from 'src/metadata/data-source/data-source.service';
+import { ObjectMetadataService } from 'src/metadata/object-metadata/object-metadata.service';
+import { WorkspaceDataSourceService } from 'src/workspace/workspace-datasource/workspace-datasource.service';
+import { ObjectMetadataHealthService } from 'src/workspace/workspace-health/services/object-metadata-health.service';
+import { FieldMetadataHealthService } from 'src/workspace/workspace-health/services/field-metadata-health.service';
+
+@Injectable()
+export class WorkspaceHealthService {
+  constructor(
+    private readonly dataSourceService: DataSourceService,
+    private readonly typeORMService: TypeORMService,
+    private readonly objectMetadataService: ObjectMetadataService,
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+    private readonly objectMetadataHealthService: ObjectMetadataHealthService,
+    private readonly fieldMetadataHealthService: FieldMetadataHealthService,
+  ) {}
+
+  async healthCheck(
+    workspaceId: string,
+    options: WorkspaceHealthOptions = { mode: WorkspaceHealthMode.All },
+  ): Promise<WorkspaceHealthIssue[]> {
+    const schemaName =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+    const issues: WorkspaceHealthIssue[] = [];
+
+    const dataSourceMetadata =
+      await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
+        workspaceId,
+      );
+
+    // Check if a data source exists for this workspace
+    if (!dataSourceMetadata) {
+      throw new NotFoundException(
+        `Datasource for workspace id ${workspaceId} not found`,
+      );
+    }
+
+    // Try to connect to the data source
+    await this.typeORMService.connectToDataSource(dataSourceMetadata);
```

Comment:
Why do we need to connect to the workspace datasource here? This is to check if this settings are valid and that we can connect later?

Reply:
Yes exactly, it's just to check if the connection to this workspace is working properly